workspaces/adventure-pack/goodies/typescript/Iterator.prototype.flatMap/index.ts

```diff
@@ -0,0 +1,34 @@
+import "../Iterator.prototype.toIterable";
+import { iteratorPrototype } from "../Iterator.prototype";
+
+declare global {
+  interface Iterator<T> {
+    flatMap<TOut>(
+      this: Iterator<T>,
+      callbackFn: (element: T, index: number) => Iterator<TOut>,
+    ): Generator<TOut, void, void>;
+  }
+}
+
+iteratorPrototype.flatMap ??= function* <TIn, TOut>(
+  this: Iterator<TIn>,
+  callbackFn: (element: TIn, index: number) => Iterator<TOut>,
+): Generator<TOut, void, void> {
+  if (typeof this !== "object") {
+    throw new TypeError("flatMap called on non-object");
+  }
```

Comment:
This shouldn't be possible, do you have an example of what this was meant to defend against?

Reply:
I was following [this pseudo code](https://tc39.es/proposal-iterator-helpers/#sec-iteratorprototype.flatmap) for inspiration, but I removed this handling